lib/phabricator/phab_object.rb

```diff
@@ -2,18 +2,126 @@
 
 module Phabricator
   class PhabObject
-    attr_reader :phid
+    def self.inherited(other)
```

Comment:
OOC, what do you mean by "this"? If you mean at a high level, all of the prop declaration stuff, I had the same thought. https://github.com/stripe-internal/chalk-dsl/ would actually fit the bill here, but it's not yet a public repo, and it really only saves a few lines of code, so it's arguably not worth dealing with an extra dependency. Most of the logic here is for the prop translation stuff, which would need to exist either way.